RELEASE_NOTES.txt

```diff
@@ -1,6 +1,120 @@
+* Improved behavior of speed-based throttling when dirty pages ratio is low.
+* Improved index-reader.sh utility: add free size calculation.
+* Improved index-reader.sh utility: analyze and output information about the actual usage of inline space in the index.
+* Improved index-reader.sh utility: exclude repeated stack traces.
+* Improved logging levels for situations when dealing with a client node.
+* Improved registered service name lookup performance and reduced service shutdown times.
+* Improved service name lookup performance and reduced service startup times.
+* Improved speed of the index-reader.sh utility.
+* Lucene library version was updated to 8.11.2.
+* Moved ignite-cloud IP Finder to the Ignite Extensions project.
+* Moved ignite-mesos to the Ignite Extensions project.
+* Moved ignite-spark modules to the Ignite Extensions project.
+* Moved ignite-yarn to the Ignite Extensions project.
+* Moved inigte-aop module to the Ignite Extensions project.
+* Removed deprecated log4j 1.x module.
+* Update Tomcat Servlet API dependency version to 9.0.63.
+* Update jsonpath dependency to 2.7.0.
+* Updated Jackson Databind dependency to 2.12.7 (CVE-2020-36518).
+* Updated Jetty dependency to 9.4.43.
+* Updated Spring dependency to 5.2.22.
+* Updated the MySql connector dependency version (fixes CVE-2021-2471, CVE-2022-21363).
+
+Java thin client:
+* Added AtomicLong.
+* Added automatic binary configuration. Name mapper and compact footer settings will be set according to cluster configuration.
+* Added IgniteSet.
+* Added partition awareness to AtomicLong - send network requests directly to the primary node for the given instance.
+* Added support of IndexQuery feature in Java thin client.
+* Fixed hang on client startup when deprecated TLS version is used.
+* Fixed race condition and potential hang on client close.
+* Fixed an issue that caused addresses are not reloaded from ClientAddressFinder on connection loss.
+
+JDBC:
+* Fixed handle BinaryObjectException on response deserialize at the thin JDBC.
+* Fixed parsing quoted delimiters in CSV fields content for SQL command COPY.
+
+.Net thin client:
+* Added AtomicLong.
+
+.Net:
+* Fixed ServiceCallContext access from async methods.
+* Fixed StringComparer serialization.
+* Fixed net461 release binaries.
+* Improved Java detection on Linux.
+
+C++ thin client:
+* Added partition property to SqlFieldsQuery for C++ thin client.
+* Added support ScanQuery for C++ thin client.
+* Added user thread pool size to public configuration of the C++ thin client.
+
+SQL:
+* Calcite-based SQL engine is now independent of H2-based SQL engine and doesn't require 'ignite-indexing' module and H2 to be in classpath anymore.
+* Added segmented indexes support for Calcite engine.
+* Added support for IN criterion for IndexQuery: IndexQuery.setCriteria(in("A", Arrays.asList(1, 2 ,3))).
+* Fixed an issue causing a wrong result to be returned if subquery with filter is used on the left shoulder of LEFT JOIN.
```

Comment:
Fixed incorrect query result if a subquery with a filter was used on the left shoulder of a LEFT JOIN.